### ISSUE TITLE
DPP overriding compiler flags

### ISSUE THREAD

**Issue by PeriodicSeizures** [brainboxdotcc/DPP#971]
Using vcpkg, where DPP is installed/built with `vcpkg install dpp --triplet=x64-windows`. Latest release is being used (v10.0.26) according to `vcpkg list`.

**Describe the bug**
Setting c++ version does not work, DPP overrides c++latest with c++17. Also see https://github.com/brainboxdotcc/DPP/issues/710.

**To Reproduce**

CMakeLists.txt: 
```cmake
cmake_minimum_required (VERSION 3.20)
project(DppTest)

find_package(dpp CONFIG REQUIRED)

add_executable(${PROJECT_NAME} "src/Main.cpp")

target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_23)

target_link_libraries(${PROJECT_NAME}
    PRIVATE dpp::dpp
)
```

src/Main.cpp:
```cpp
#include <iostream>

// concept not being recognized
template<typename T> requires std::is_integral_v<T>
struct Something {};

int main(int argc, char **argv) {
    std::cout << "Test\n";

    Something<int> T{};

    return 0;
}

```

**Expected behavior**
Program should compile.

**Comment by Mishura4** [comment]
@braindigitalis Did we ever push #863 to vcpkg?

**Comment by Mishura4** [comment]
Issue probably caused by
https://github.com/microsoft/vcpkg/pull/34185/files#diff-d66ae78a65c40c62e98e7181dbe7a4dbf53444f46e361867ddc35f6794d7dfe6L10

**Comment by Mishura4** [comment]
So, the problem with our VCPKG CMakeLists:

`target_compile_definitions` cannot contain flags. They should be only used for preprocessor macros.
`target_compiler_options`, while, being the correct command for flags, does propagate flags to a consumer if PUBLIC, which in the case of the cmake standard probably shouldn't be the case.

The correct and most portable way to change the c++ standard is to juse use
`target_compile_features(dpp PUBLIC cxx_std_17)`. This will allow users to overwrite it with a more recent standard.

@braindigitalis I'll let you handle this since you have more experience with vcpkg than I do.

**Comment by Mishura4** [comment]
Potential fix #973 merged into master
